### PR TITLE
ITL-1-Basic API for ToDoList with errors and CRUD methods

### PR DESCRIPTION
**Pull request by Barabasbalazs**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -2,16 +2,25 @@ import express from "express";
...
 app.use(express.json());
 app.use(express.urlencoded({ extended: false }));
 
-app.use("/", router);
+app.use("/api", router);
+
+app.use(handleError);
 
-const port = 8080;
+const port = process.env.PORT;
```
Comment (SzaszWebgurus) [inline src/index.ts]: add default value

File: src/models/todo-model.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import mongoose, { Document, Schema } from "mongoose";
+
+export interface ToDo extends Document {
+  title: { type: string; required: true };
```
Comment (roli-wg) [inline src/models/todo-model.ts]: Using interfaces every field is required by default you don't have to explicitly specify

File: src/routes/todos/todo-controller.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import { Request, Response, NextFunction } from "express";
+import { ToDo } from "../../models/todo-model";
+import { todoService } from "../../services/todo-service";
+import { errors } from "../../utils/errors";
+
+export const todoController = {
+  saveToDo: async (
+    req: Request<{}, {}, ToDo>,
+    res: Response,
```
Comment (roli-wg) [inline src/routes/todos/todo-controller.ts]: Is a good practice to specify also the Response type `Response<Todo>`

File: src/routes/todos/todo-controller.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      if (!newToDo) {
+        return next(errors.unknown);
+      }
+      res.status(201).json(newToDo);
+    } catch (e) {
+      return next(errors.unknown);
+    }
+  },
+
+  getAllToDos: async (req: Request, res: Response, next: NextFunction) => {
```
Comment (roli-wg) [inline src/routes/todos/todo-controller.ts]: Try to keep the naming simple you are inside the todoController so you don't need to add todo/todos as sufix for your methods 
You can name just save, getAll, getById...

File: src/routes/todos/todo-controller.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    }
+  },
+
+  updateToDo: async (
+    req: Request<{ _id: string }, {}, Partial<ToDo>>,
+    res: Response,
+    next: NextFunction
+  ) => {
+    try {
+      console.log(req.params._id);
```
Comment (roli-wg) [inline src/routes/todos/todo-controller.ts]: Please remove all console.log

File: src/services/todo-service.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import model, { ToDo } from "../models/todo-model";
+
+export const todoService = {
+  insertToDo: async (toDo: ToDo): Promise<ToDo | void> => {
```
Comment (roli-wg) [inline src/services/todo-service.ts]: Pay attention to naming here as well

File: src/services/todo-service.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    return await model.findById(id);
+  },
+
+  removeToDo: async (id: string): Promise<ToDo | null> => {
+    return await model.findByIdAndDelete(id);
+  },
+
+  updateToDo: async (id: string, toDo: Partial<ToDo>): Promise<ToDo | null> => {
+    return await model.findOneAndUpdate({ _id: id }, toDo, {
+      returnOriginal: false,
```
Comment (roli-wg) [inline src/services/todo-service.ts]: use `new:true` as option for update to return the updated document

**Review by roli-wg — CHANGES_REQUESTED** [review]

**Review by roli-wg — APPROVED** [review]

**Pull request merged**